fish-rust/src/env/mod.rs

```diff
@@ -5,4 +5,49 @@ pub mod var;
 
 pub use env_ffi::EnvStackSetResult;
 pub use environment::*;
+use std::ffi::CString;
+use std::sync::atomic::{AtomicBool, AtomicUsize};
 pub use var::*;
+
+/// Limit `read` to 100 MiB (bytes, not wide chars) by default. This can be overriden with the
+/// `fish_read_limit` variable.
+pub const DEFAULT_READ_BYTE_LIMIT: usize = 100 * 1024 * 1024;
+
+/// The actual `read` limit in effect, defaulting to [`DEFAULT_READ_BYTE_LIMIT`] but overridable
+/// with `$fish_read_limit`.
+#[no_mangle]
+pub static READ_BYTE_LIMIT: AtomicUsize = AtomicUsize::new(DEFAULT_READ_BYTE_LIMIT);
+
+/// The curses `cur_term` TERMINAL pointer has been set up.
+#[no_mangle]
+pub static CURSES_INITIALIZED: AtomicBool = AtomicBool::new(false);
+
+/// Does the terminal have the "eat new line" glitch.
+#[no_mangle]
+pub static TERM_HAS_XN: AtomicBool = AtomicBool::new(false);
+
+mod ffi {
+    extern "C" {
+        pub fn setenv_lock(
+            name: *const libc::c_char,
+            value: *const libc::c_char,
+            overwrite: libc::c_int,
+        );
+        pub fn unsetenv_lock(name: *const libc::c_char);
```

Comment:
these should be private, same for other extern C decls below

Reply:
No, actually. They're in a submodule so making them `pub` makes them visible to `env/mod.rs` itself. We'd have to `pub use self::ffi::*` to actually make them public to the crate.